database/src/main/java/com/firebase/ui/database/FirebaseIndexRecyclerAdapter.java

```diff
@@ -59,8 +59,11 @@
  * </pre>
  *
  * @param <T>  The Java class that maps to the type of objects stored in the Firebase location.
- * @param <VH> The ViewHolder class that contains the Views in the layout that is shown for each object.
+ * @param <VH> The ViewHolder class that contains the Views in the layout that is shown for each
+ *             object.
+ * @deprecated use {@link com.firebase.ui.database.adapter.FirebaseIndexRecyclerAdapter} instead
  */
+@Deprecated
```

Comment:
Sorry for the delay here but I have been trying to think of some way out of this situation, been playing with your branch in my IDE.

We need to find *some* way to avoid deprecating `FirebaseRecyclerAdapter` and `FirebaseListAdapter` pretty much at all costs.  99% of our users just want to use the adapters as-is, but we'd be breaking all of them for the 1% who want to extend the adapters in interesting ways.  I see the issue in the changing of `protected onDataChanged` to `public onDataChanged`, that's what we need to get around.

Here's a random idea:
  * Rename the method in `ChangeEventListener` to something else, have that method delegate to `onDataChanged` in the default implementation of each adapter.
  * Deprecate the `onDataChanged` method and point people to use the newly named method.

Would that enable us to keep the adapters in the original package and not deprecated?  I'm sure I'm missing something, hard to reason about these things in a comment box.